capstone/src/main/java/com/google/sps/servlets/FollowServlet.java

```diff
@@ -62,6 +63,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
             room.removeFollower(user);
         }
 
+        DatabaseService.save(user.getUserEntity());
         DatabaseService.save(room.getRoomEntity());
```

Comment:
As a note: this right here is totally correct given the context but a little unfortunate. Because calling room.addFollower actually updates the user itself, you have to remember to save this. I might in another context not realize that adding a follower actually modifies the user as well due to caching and not realize I need to save the user. This can be a source of bugs and problematic conceptually, and is a huge benefit of using an ORM of some kind (thats what its named for SQL dbs, idk what the NoSQL equivalent is called), which automatically does the Object <-> Database mapping and save calls and everything and lets you just treat the documents/rows in a database as though they are just your model objects.

Reply:
Got it. That's an interesting concept, I will look more into it.